pkg/zonemgr/zone_file_cache.go

```diff
@@ -40,10 +39,16 @@ type ZoneFileCache struct {
 	vmiRecordsMap map[string][]string
 }
 
-func NewZoneFileCache(nameServerIP string, domain string) *ZoneFileCache {
+func NewZoneFileCache(nameServerIP string, domain string, soaSerial *int) *ZoneFileCache {
+	soaSerialInt := 0
```

Comment:
soaSerialInt better to just use the argument soaSerial as is, and it will have (does it already ? if not please check) the value zero if the fetch didnt find a previous serial

Reply:
the idea is to handle nil value if ZoneFile not exist

Reply:
but 0 is good enough, as long as there is no error, we don't need to differentiate between nil and 0

Reply:
https://github.com/kubevirt/kubesecondarydns/pull/45#discussion_r1054124649